pygitguardian/client.py

```diff
@@ -264,15 +264,15 @@ def get(
     def post(
         self,
         endpoint: str,
-        data: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
+        json: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
```

Comment:
This change makes sense, but it's a breaking change. Code like this:

```
client.post(endpoint, data=something)
```

Won't work anymore. We can handle this by bumping the major to 2, but it's always nicer to avoid doing so.

My understanding is we would need the `data` argument to be able to post non-JSON data. Is there any reason we need it to not be JSON?

Reply:
Indeed, the idea was to change every occurrence of client.post in the code to replace `data=something` with `json=something`.

If the risk is too great, maybe the best solution is to add an argument to GGClient.post :
`additional_data: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,`

and then in `self.request` add `data=additional_data`.

It won't break existing code but it is slightly less clear. (IMO having a data argument used for json is not intuitive)

This would give something like this :

```
    def post(
        self,
        endpoint: str,
        data: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
        additional_data: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
        version: str = DEFAULT_API_VERSION,
        extra_headers: Optional[Dict[str, str]] = None,
        **kwargs: Any,
    ) -> Response:
        return self.request(
            "post",
            endpoint=endpoint,
            json=data,
            version=version,
            extra_headers=extra_headers,
            data=additional_data
            **kwargs,
        )
```

Reply:
I also noticed I did not updated  the new method `GGClient.create_jwt` (change `data=` to `json=`)
Based on the final decision i will either :

- commit the change to `GGClient.create_jwt`
- commit a change to revert previous change and add `additionnal_data` to `GGClient.post`

Reply:
> IMO having a data argument used for json is not intuitive

I 100% agree with you on this. We would need to change this when we break API.

> I also noticed I did not updated the new method GGClient.create_jwt (change data= to json=)

That worries me a bit: I would have expected the tests to catch the problem. Going to investigate.

I just thought about an alternative fix, which would not break compatibility: change `iac_directory_scan()` to call `client.request("post", endpoint=..., data=...)` instead of calling `client.post(endpoint=..., data=...)`. What do you think?